File: packages/shared-types/src/document.ts
export type DocumentStatus = "pending" | "processing" | "completed" | "failed";
export type FileType = "pdf" | "docx" | "txt" | "md";

export interface Document {
  id: string;
  name: string;
  original_name: string;
  file_type: FileType;
  file_size: number;
  file_hash: string;
  status: DocumentStatus;
  chunk_count: number;
  page_count: number;
  error_message?: string;
  metadata: Record<string, unknown>;
  created_at: string;
  updated_at: string;
}

export interface DocumentChunk {
  id: string;
  document_id: string;
  chunk_index: number;
  content: string;
  page_number?: number;
  token_count?: number;
  metadata: Record<string, unknown>;
  created_at: string;
}

export interface DocumentUploadResponse {
  id: string;
  name: string;
  status: DocumentStatus;
  message: string;
}

export interface DocumentListResponse {
  documents: Document[];
  total: number;
  page: number;
  page_size: number;
}

export interface ProcessingStatus {
  document_id: string;
  status: DocumentStatus;
  progress: number;
  current_step: string;
  error_message?: string;
}

File: apps/api/src/app/schemas/document.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class DocumentResponse(BaseModel):
    id: UUID
    name: str
    original_name: str
    file_type: str
    file_size: int
    file_hash: str
    status: str
    chunk_count: int
    page_count: int
    error_message: str | None = None
    metadata: dict = {}
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class DocumentUploadResponse(BaseModel):
    id: UUID
    name: str
    status: str
    message: str


class DocumentListResponse(BaseModel):
    documents: list[DocumentResponse]
    total: int
    page: int
    page_size: int


class ChunkResponse(BaseModel):
    id: UUID
    document_id: UUID
    chunk_index: int
    content: str
    page_number: int | None = None
    token_count: int | None = None
    metadata: dict = {}
    created_at: datetime

    model_config = {"from_attributes": True}
